src/realm/object-store/results.cpp

```diff
@@ -116,15 +116,17 @@ bool Results::is_valid() const
         m_realm->verify_thread();
     }
 
-    if (m_collection)
```

Comment:
Ah, OK, I see, the fix was put in the wrong part of the code. `import_copy_into_realm` needs to check if the collection has gone away... interesting...